### PR TITLE
Feature/new services

### PR DESCRIPTION
**Pull request by doronz88**

### CODE REVIEW

File: pymobiledevice3/remote/xpc_message.py [hunk 1]
```diff
@@ -152,6 +153,10 @@ def _decode_xpc_data(xpc_object) -> bytes:
     return xpc_object.data
 
 
+def _decode_xpc_date(xpc_object) -> datetime:
+    return datetime.fromtimestamp(xpc_object.data / 1000000000)
```
Comment (matan1008) [inline pymobiledevice3/remote/xpc_message.py]: Hard to count the 0, maybe add comment "Convert from nanoseconds to seconds"

File: pymobiledevice3/remote/remote_service.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        if output is None:
+            raise CoreDeviceError(f'Failed to invoke: {feature_identifier}. Got error: {response}')
+        return response['CoreDevice.output']
+
+    def __enter__(self):
+        self.connect()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.service.close()
```
Comment (matan1008) [inline pymobiledevice3/remote/remote_service.py]: self.close

File: pymobiledevice3/remote/remote_service.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            'CoreDevice.coreDeviceVersion': {
+                'components': [XpcUInt64Type(325), XpcUInt64Type(3), XpcUInt64Type(0),
+                               XpcUInt64Type(0), XpcUInt64Type(0)],
+                'originalComponentsCount': XpcInt64Type(2),
+                'stringValue': '325.3'},
+            'CoreDevice.deviceIdentifier': '7454ABFD-F789-4F99-9EE1-5FB8F7035ECE',
+            'CoreDevice.featureIdentifier': feature_identifier,
+            'CoreDevice.input': input_,
+            'CoreDevice.invocationIdentifier': '14A17AB8-0576-4E73-94C6-C0282A4F66E3'})
+        output = response.get('CoreDevice.output')
```
Comment (matan1008) [inline pymobiledevice3/remote/remote_service.py]:
```
if 'CoreDevice.output' not in response:
        raise CoreDeviceError(f'Failed to invoke: {feature_identifier}. Got error: {response}')
return response['CoreDevice.output']
```

or 
```
output = response.get('CoreDevice.output')
if output is None:
        raise CoreDeviceError(f'Failed to invoke: {feature_identifier}. Got error: {response}')
return output
```

File: pymobiledevice3/remote/xpc_message.py (hunk 1 above)
Comment (doronz88) [reply pymobiledevice3/remote/xpc_message.py]: done

File: pymobiledevice3/remote/remote_service.py (hunk 2 above)
Comment (doronz88) [reply pymobiledevice3/remote/remote_service.py]: done

File: pymobiledevice3/remote/remote_service.py (hunk 3 above)
Comment (doronz88) [reply pymobiledevice3/remote/remote_service.py]: done

**Pull request closed without merging**